lib/active_merchant/connection.rb

```diff
@@ -76,7 +76,7 @@ def request(method, body, headers = {})
               http.get(endpoint.request_uri, headers)
             when :post
               debug body
-              http.post(endpoint.request_uri, body, RUBY_184_POST_HEADERS.merge(headers))
+              http.post(endpoint.request_uri, body, headers.reverse_merge!(RUBY_184_POST_HEADERS))
```

Comment:
Is this change related to the HTTP version ❓

Reply:
Why using reverse merge? Is order of these headers also important?